src/xbot/common/controls/sensors/Gamepad.java

```diff
@@ -1,52 +1,114 @@
-        rightTrigger = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.RightTrigger);
-    }
-
-    @Override
-    public XJoystick getLeftStick() {
-        return leftStick;
+        internalGamepad = new XboxController(port);
     }
 
     @Override
-    public XJoystick getRightStick() {
-        return rightStick;
-    }
+    //Joysticks---------------------------------------------------------------------------------------------
+   public double getLeftStickX(){
+       
+       return internalGamepad.getX(Hand.kLeft);
+   }
     
-    @Override
-    public XJoystick getDpad() {
-        return dpad;
+    public double getRightStickX(){
+        
+        return internalGamepad.getX(Hand.kRight);
     }
+    
+   public double getLeftStickY(){
+       
+       return internalGamepad.getY(Hand.kLeft);
+   }
+   
+   public double getRightStickY(){
+       
+       return internalGamepad.getY(Hand.kRight);
+   }
+   
+   //Triggers-----------------------------------------------------------------------------------------------
+   public double getLeftTriggerAxis(){
+       
+       return internalGamepad.getTriggerAxis(Hand.kLeft);
+   }
+   
+   public double getRightTriggerAxis(){
+       
+       return internalGamepad.getTriggerAxis(Hand.kRight);
+   }
 
-    @Override
-    public XJoystick getLeftTrigger() {
-        return leftTrigger;
-    }
+   //Bumpers------------------------------------------------------------------------------------------------
+   public boolean getLeftBumper() {
+       
+       return internalGamepad.getBumper(Hand.kLeft);
+   }
+   
+  public boolean getRightBumper() {
+       
+       return internalGamepad.getBumper(Hand.kRight);
+  }
+  
+  //Buttons-------------------------------------------------------------------------------------------------
+  public boolean getAButton(){
+      
+      return internalGamepad.getAButton();
+  }
+  
+  public boolean getBButton(){
+      
+      return internalGamepad.getBButton();
+  }
+  
+  public boolean getXButton(){
+      
+      return internalGamepad.getXButton();
+  }
+  
+  public boolean getYButton(){
+      
+      return internalGamepad.getYButton();
+  }
+  
+  public boolean getBackButton(){
+      
+      return internalGamepad.getBackButton();
+  }
+  
+  public boolean getStartButton(){
+      
+      return internalGamepad.getStartButton();
+  }
+  
+  //Sticks---------------------------------------------------------------------------------------------------
+  public boolean getLeftStickButton(){
+      
+      return internalGamepad.getStickButton(Hand.kLeft);
+  }
+  
+  public boolean getRightStickButton(){
+      
+      return internalGamepad.getStickButton(Hand.kRight);
+  }
 
-    @Override
-    public XJoystick getRightTrigger() {
-        return rightTrigger;
-    }
+  public XYPair getLeftStick(){
+      return new XYPair(getRightStickX(), getLeftStickX());
```

Comment:
It looks like these two methods are inverted partially; each should return the components of a particular stick, not one from each.